docs/source/fundamentals/benefits.md

```diff
@@ -0,0 +1,133 @@
+
+With REST, you would have to call all of these endpoints for each item in the list, filter down the data you need, and aggregate all of the remaining data into the shape your components consume.
+
+```bash
+GET /api/dogs/breeds
+GET /api/dogs/images
+GET /api/dogs/activities
+```
+
+Not only is this approach time-consuming, it's also prone to error and difficult to reuse logic across platforms. Compare this with GraphQL's declarative way to query data:
+
+```graphql
+const GET_DOGS = gql`
+  query {
+    dogs {
+      id
+      breed
+      image {
+        url
+      }
+      activities {
+        name
+      }
+    }
+  }
+`;
+```
+
+Here, we're describing the shape of the object we want to receive from the server. GraphQL takes care of combining and filtering the data while returning exactly what we ask for.
+
+How do we use this query in our app? Apollo Client builds off of GraphQL's declarative approach to data fetching. In a React app, all of the logic for retrieving your data, tracking loading and error states, and updating your UI is encapsulated in a single Query component. This encapsulation makes composing your data fetching components with your presentational components a breeze! Let’s see how to fetch GraphQL data with Apollo Client in a React app:
+
+```jsx
+const Feed = () => (
+  <Query query={GET_DOGS}>
+    {({ loading, error, data }) => {
+      if (error) return <Error />
+      if (loading || !data) return <Fetching />;
+
+      return <DogList dogs={data.dogs} />
+    }}
+  </Query>
+);
+```
+
+Apollo Client takes care of the request cycle from start to finish, including tracking loading and error states for you. There’s no middleware to set up or boilerplate to write before making your first request, nor do you need to worry about transforming and caching the response. All you have to do is describe the data your component needs and let Apollo Client do the heavy lifting. 💪
+
+You’ll find that when you switch to Apollo Client, you’ll be able to delete a lot of unnecessary code related to data management. The exact amount will vary depending on your application, but some teams have reported up to thousands of lines. To learn more about how Apollo Client enables advanced features like optimistic UI, refetching, and pagination with less code, check out our guide on [state management](../guides/state-mgmt.html).
+
+<h2 id="performance">Improved performance</h2>
+
+In many cases, layering a GraphQL API over your existing REST endpoints can improve your app's performance, especially on devices with slow network connections. While you should always measure to determine how integrating GraphQL will affect your application, it's generally accepted that GraphQL improves performance by helping avoid round trips to the server and reducing payload size.
+
+<h3 id="payload">Smaller payloads</h3>
```

Comment:
I understand and appreciate short(er) named anchors, but I don't think they should lose their clarity.  For example, from a maintenance standpoint if I had to look at `payload` in a list of, say, redirects, I'd rather it be called `smaller-payloads`, which is what this anchor would be if it the markdown renderer generated it from `##` syntax.  I feel that anyone using such anchor links elsewhere might also appreciate that, and it reduces the possibility of a conflict in the future (when we bring in, say, `## Compressed payloads` 🤷‍♂️)

Reply:
That's a really good point, I'll switch it to `smaller-payloads` for clarity. Also, I totally didn't realize that the markdown renderer autogenerates ids from `##` syntax! Thanks for the tip 😀